src/core/numerics/interpolator/interpolator.hpp

```diff
@@ -485,33 +485,36 @@ class Interpolator : private Weighter<interpOrder>
         // calculated twice, and not for each E,B component.
 
         PHARE_LOG_START("MeshToParticle::operator()");
-        for (auto currPart = begin; currPart != end; ++currPart)
-        {
-            auto& iCell = currPart->iCell;
-            auto& delta = currPart->delta;
-            indexAndWeights_<QtyCentering, QtyCentering::dual>(layout, iCell, delta);
-            indexAndWeights_<QtyCentering, QtyCentering::primal>(layout, iCell, delta);
-
-            auto indexWeights = std::forward_as_tuple(dual_startIndex_, dual_weights_,
-                                                      primal_startIndex_, primal_weights_);
-
-            currPart->Ex
-                = meshToParticle_.template operator()<GridLayout, Scalar::Ex>(Ex, indexWeights);
-            currPart->Ey
-                = meshToParticle_.template operator()<GridLayout, Scalar::Ey>(Ey, indexWeights);
-            currPart->Ez
-                = meshToParticle_.template operator()<GridLayout, Scalar::Ez>(Ez, indexWeights);
-            currPart->Bx
-                = meshToParticle_.template operator()<GridLayout, Scalar::Bx>(Bx, indexWeights);
-            currPart->By
-                = meshToParticle_.template operator()<GridLayout, Scalar::By>(By, indexWeights);
-            currPart->Bz
-                = meshToParticle_.template operator()<GridLayout, Scalar::Bz>(Bz, indexWeights);
-        }
+
+        auto& iCell = currPart.iCell;
+        auto& delta = currPart.delta;
+        indexAndWeights_<QtyCentering, QtyCentering::dual>(layout, iCell, delta);
+        indexAndWeights_<QtyCentering, QtyCentering::primal>(layout, iCell, delta);
+
+        auto indexWeights = std::forward_as_tuple(dual_startIndex_, dual_weights_,
+                                                  primal_startIndex_, primal_weights_);
+
+        E_B_tuple particle_EB;
+        auto& [pE, pB]        = particle_EB;
+        auto& [pEx, pEy, pEz] = pE;
+        auto& [pBx, pBy, pBz] = pB;
+
+        auto const& [Ex, Ey, Ez] = Em.E();
+        auto const& [Bx, By, Bz] = Em.B();
+
+        pEx = meshToParticle_.template operator()<GridLayout, Scalar::Ex>(Ex, indexWeights);
```

Comment:
it is possible, we would just need to use an actual function and not `operator()`

which I would like to do anyway in interpolator to avoid potential collisions 

I think it's a bit strange that we do different things for `operator()` depending on the number/type of inputs

Reply:
so do we change something now or later?

Reply:
I have updated this on my [SOA](https://github.com/PhilipDeegan/PHARE/blob/mkn/src/core/numerics/interpolator/interpolator.hpp#L522C9-L522C85) branch, but we can do something now too

it could also be like 
```
pEx = meshToParticle_.template op<Scalar::Ex>(Ex, indexWeights, layout);
```

but I think this might have some minor runtime difference, as the layout would be passed unused to the operating function